### ISSUE TITLE
Switch from Unlicense to CC0

### ISSUE THREAD

**Issue by emmatebibyte** [rmyorston/pdpmake#55]
The Unlicense doesn’t dedicate the source code to the public domain in all jurisdictions. It also is not as battle-tested as CC0.

https://chrismorgan.info/blog/unlicense/
https://lists.opensource.org/pipermail/license-review_lists.opensource.org/2012-January/001386.html

**Comment by averyterrel** [comment]
I agree that switching to a more "real" license would be preferable. 0BSD may be preferable to CC0, since it is significantly shorter and more readable.

**Comment by emmatebibyte** [comment]
On 2024-09-08 01:05:45 -0700, "Avery" ***@***.***> wrote:

> I agree that switching to a more "real" license would be preferable.
> 0BSD may be preferable to CC0, since it is significantly shorter and
> more readable.

Does 0BSD have the same global fallback features as CC0 does for
countries without or with different public domain systems?

-- 
Emma Tebibyte (fae/faer)
http://tebibyte.media/~emma

**Comment by averyterrel** [comment]
It doesn't contribute the work to the public domain (explicitly) at all, it just gives "[permission] to use, copy, modify, and/or distribute this software for any purpose with or without fee".

**Comment by emmatebibyte** [comment]
On 2024-09-29 20:08:53 -0700, "Avery" ***@***.***> wrote:

> It doesn't contribute the work to the public domain (explicitly) at
> all, it just gives "[permission] to use, copy, modify, and/or
> distribute this software for any purpose with or without fee".
> See: https://choosealicense.com/licenses/0bsd/

I suppose, then, that it is up to @rmyorston as to which of the two is
preferable (if the license is to be changed).

-- 
Emma Tebibyte (fae/faer)
http://tebibyte.media/~emma

**Comment by rmyorston** [comment]
I don't have an informed opinion on the relative merits of the licences mentioned here.

My preference is to take no action.

**Comment by averyterrel** [comment]
Unlicense, in my unprofessional opinion, would not hold up in jurisdictions where Public Domain is not recognized

**Comment by mechite** [comment]
> I don't have an informed opinion on the relative merits of the licences mentioned here.
> My preference is to take no action.

@rmyorston: unrelated, but I find your current license similar to Unlicense, but currently I cannot use the SPDX license identifier "Unlicense" as pdpmake lacks this clause:

```
In jurisdictions that recognize copyright laws, the author or authors of this
software dedicate any and all copyright interest in the software to the public
domain.  We make this dedication for the benefit of the public at large and to
the detriment of our heirs and successors. We intend this dedication to be an
overt act of relinquishment in perpetuity of all present and future rights to
this software under copyright law.
```

You are almost word-for-word with [Unlicense-libwhirlpool](https://spdx.org/licenses/Unlicense-libwhirlpool.html), but your disclaimer clause has not been trimmed from the original Unlicense like they did. &nbsp; So I, a user of SPDX, have issues due to that. &nbsp; However, the license itself is fine.

> It also is not as battle-tested as CC0.

IANAL. &nbsp; To be frank, for the same reasons as ISC holds up, due to the Berne convention, the Unlicense is a perfectly usable license and has a normal permission grant just like any BSD/ISC/MIT/CC0/etc. &nbsp; Google banned use of it internally (iirc), probably due to the lack of any severability clause (for the public domain notice). &nbsp; Ron, who had the intent of releasing the rights to this software, will not be able to maliciously litigate, and the license is perfectly acceptable in the real world.

**Comment by ale5000-git** [comment]
@rmyorston

In doubt, just take both :-)

If you want you can also use this:
`// SPDX-License-Identifier: Unlicense OR CC0-1.0`

It means that the user can follow alternatively `Unlicense` or `CC0-1.0`.
"OR" mean alternatively (so maximum freedom to choose one or another), while "AND" mean both at the same time (so restricted to both license terms at the same time).
